core/templates/domain/learner_dashboard/learner-dashboard-activity-backend-api.service.spec.ts

```diff
@@ -239,7 +239,6 @@ describe('Learner playlist Backend Api service ', () => {
         untracked_topic_ids: []
       });
     const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
-      setTimeout(opt.beforeDismiss);
```

Comment:
why remove these?

Reply:
I don't really find a use of these statements, these throw an error but frontend tests pass even without them. Why do we exactly need to do `opt.beforeDismiss` while opening an ngbModal ? There use is defined here https://darchuk.net/2019/01/18/unit-testing-ngbmodal-modal-service/ .

Reply:
hmm.. maybe the other reviewers can assess this better

Reply:
Tests seem to pass for the last three commits made in the PR. @srijanreddy98 any thoughts on this ? If not removed we'll have to introduce a check everywhere for them to be not undefined
```
if (opt?.beforeDismiss !== undefined) {
        setTimeout(opt.beforeDismiss);
      }
```